pkg/install/genevalogging.go

```diff
@@ -0,0 +1,399 @@
+	  key MESSAGE
+	  pattern .+
+	  tag journald
+	</rule>
+  </match>
+  <filter journald>
+	@type record_transformer
+	enable_ruby true
+	<record>
+	  MESSAGE ${record["MESSAGE"].nil? ? nil : record["MESSAGE"].force_encoding("UTF-8").encode("ASCII", invalid: :replace, undef: :replace)}
+	  # data format:
+	  # k8s_apiserver_apiserver-vqxg4_kube-service-catalog_72ce4d73-5224-11e9-98d0-000d3a196756_0
+	  CONTAINER ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[1] }
+	  POD ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[2] }
+	  NAMESPACE ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[3] }
+	  CONTAINER_ID ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[4] }
+	</record>
+  </filter>
+  <match **>
+	@type mdsd
+	acktimeoutms 0
+	buffer_type memory
+	buffer_queue_full_action block
+	disable_retry_limit true
+	djsonsocket /var/run/mdsd/default_djson.socket
+	emit_timestamp_name time
+	flush_interval 10s
+  </match>
+`
+)
+
+func mdsdConfig(location, resourceGroup, subscriptionID, clusterName, genevaLoggingNamespace, genevaLoggingAccount string) (string, error) {
+	var tmpl = `<?xml version="1.0" encoding="utf-8"?>
+	    <MonitoringManagement version="1.0" namespace="{{ .Namespace | Escape }}" eventVersion="1" timestamp="2017-08-01T00:00:00.000Z">
+	        <Accounts>
+	            <Account moniker="{{ .Account | Escape }}" isDefault="true" autoKey="false"/>
+	        </Accounts>
+	        <Management eventVolume="Large" defaultRetentionInDays="90">
+	            <Identity tenantNameAlias="ResourceName" roleNameAlias="ResourceGroupName" roleInstanceNameAlias="SubscriptionId">
+	                <IdentityComponent name="Region">{{ .Region | Escape }}</IdentityComponent>
+	                <IdentityComponent name="SubscriptionId">{{ .SubscriptionId | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceGroupName">{{ .ResourceGroupName | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceName">{{ .ResourceName | Escape }}</IdentityComponent>
+	            </Identity>
+	            <AgentResourceUsage diskQuotaInMB="50000"/>
+	        </Management>
+	        <Sources>
+	            <Source name="journald" dynamic_schema="true"/>
+	            <Source name="audit" dynamic_schema="true"/>
+	        </Sources>
+	        <Events>
+	            <MdsdEvents>
+	                <MdsdEventSource source="journald">
+	                    <RouteEvent eventName="CustomerSyslogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	                <MdsdEventSource source="audit">
+	                    <RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	            </MdsdEvents>
+	        </Events>
+		</MonitoringManagement>`
+
+	t := template.Must(template.New("").Funcs(map[string]interface{}{
```

Comment:
using `template.Must` is good, but only do it at package scope, so that it is called at package initialisation time and hence if there's a panic, it happens immediately on startup